### PR TITLE
Dashboard: POC to run existing e2e with `dashboardScene` feature toggle

### PR DESCRIPTION
**Pull request by axelavargas**
#### Depends on this PR ([scenes PR](https://github.com/grafana/scenes/pull/648))

This POC PR is part of Proposal 3

It is possible to run our e2e with a feature flag enabled by default.  This PR enables the `dashboardScene` feature toggle (using local storage) and sets up an extra `scenes` option on the cypress config.  The idea will be to run this "scene" e2e tests in a scheduled github action agains main branch to prevent increasing CI times.

Explorations on this proposal: 
- #### Use the same e2e tests and modify them based on the feature toggle (failed)
	- I tried to standardize e2e selectors across different flows (adding a dashboard)
	- Blocker: Some dashboard flows involve editing, it became cumbersome to achieve this, due to the new edit mode we added. I found myself adding a lot of conditionals for the existing e2e code to handle these cases
- #### Duplicate e2e tests and refactor them to only align with the new flows (the changes on this PR) 
	- Created a new folder called `scenes` in the `e2e` folder and duplicated some of the suites that have dashboard logic 
	- Added an extra option called `scenes` on the cypress config, it will only run the e2e that are on that `scenes` folder
	- Added a new command to enable the feature toggle via local storage 
	- Started to refactor all the tests and flows to accommodate the new UX flow (the edit mode)

##### E2E tests migrated
- `smoke-tests-suite`
	- [x] 1-smoketests
	- [x] panels_smokecreen
- `dashboard-suite`
	- [x] import-dashboard.spec.ts 
	- [x] embedded-dashboard.spec.ts
	- [x] dashboard-browse.spec.ts   
	- [x] dashboard-live-streaming.spec.ts 
	- [x] dashboard-timepicker.spec.ts     
##### Todo
- `dashboard-suite`
	- [x] Repeating_a_panel_horizontally.spec.ts  ----> This is not rendered with scenes? (need research) (only empty in screenshot)
	- [x]  Repeating_a_panel_vertically.spec.ts ----> This is not rendered with scenes? (need research) (only empty in screenshot)
	- [x] Repeating_an_empty_row.spec.ts
	- [x] dashboard-browse-nested.spec.ts ---> is it empty? no output when running (only empty in screenshot)
	- [x] dashboard-public-create.spec.ts 
	- [x] dashboard-public-templating.spec.ts   
	- [x] dashboard-templating.spec.ts
	    - Should: var-Server=All | Is: var-Server=A%27A%22A&var-Server=BB%5CB&var-Server=CCC difference
	- [ ] dashboard-time-zone.spec.ts   
	    - skipped due to https://github.com/grafana/grafana/issues/86420
	- [x] general-dashboards.spec.ts
	    - Does not scroll back to panel 50    
	- [x] load-options-from-url.spec.ts
	- [x] new-constant-variable.spec.ts 
	- [x] new-custom-variable.spec.ts
	- [x] new-datasource-variable.spec.ts 
	- [ ] new-interval-variable.spec.ts   ---> failed, bug found https://github.com/grafana/grafana/issues/84727 
	- [x] new-query-variable.spec.ts 
	- [x] new-text-box-variable.spec.ts      
	- [x] set-options-from-ui.spec.ts   
	- [x] templating-dashboard-links-and-variables.spec.ts
	   - uncertainty with https://github.com/grafana/grafana/issues/86437
	- [ ] textbox-variables.spec.ts 
	  - skipped due to https://github.com/grafana/grafana/issues/86435

There are number of improvements that need to happen after this PR:

- [ ] Refactor tests (quite a few places would benefit from extra e2e selectors)
- [ ] improve workflow (include enterprise?)
- [ ] migrate remaining test suites

### CODE REVIEW

**Comment by Sergej-Vlasov** [comment]
Note: some weird addition in `pkg/build/wire/` after merging with main. WIll adjust this first thing next week

File: .github/workflows/run-scenes-e2e.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Run dashboard scenes e2e
+
```
Comment (Sergej-Vlasov) [inline .github/workflows/run-scenes-e2e.yml:2]: Previous test runs of this workflow https://github.com/grafana/grafana/actions/runs/8829700520/job/24241279325?pr=84598

**Review by axelavargas — COMMENTED** [review]
@Sergej-Vlasov fantastic job implementing this one 🥇 , the refactored test and new Github action look good to me. 

I am sharing this with the @grafana/dashboards-squad members to get an approval. But I think we are good to go!

**Review by Sergej-Vlasov — APPROVED** [review]
Approving to accelerate rollout of dashboard scenes e2e tests. 
P.S. Github workflow logic was reviewed and approved [slack thread](https://raintank-corp.slack.com/archives/C0671T2J5CH/p1714553463012119)

**Pull request merged**